MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InputController.h

```diff
@@ -245,29 +245,59 @@ class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerDraw
 /**
     Controller for erasing peaks on an unwrapped surface.
   */
-class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerErase
+class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerSelection
     : public InputControllerDraw {
   Q_OBJECT
 
 public:
-  InputControllerErase(QObject *parent);
-  ~InputControllerErase() override;
+  InputControllerSelection(QObject *parent);
+  ~InputControllerSelection() override;
   void onPaint(QPainter &) override;
 
-signals:
-  void erase(const QRect &);
-
 private:
   void drawCursor(QPixmap *cursor) override;
-  void signalLeftClick() override;
   void setPosition(const QPoint &pos) override;
   void resize() override;
 
-  QRect m_rect;
+protected:
```

Comment:
It's best practice to avoid protected data members where possible - can we keep these private? Protected data increases the number of places that could possibly modify the state and cause trouble: http://softwareengineering.stackexchange.com/questions/162643/why-is-clean-code-suggesting-avoiding-protected-variables

For `m_image`, can be add a constructor parameter passing in the QPixmap for the icon? As far as I can see it is not used anywhere else?

For `m_rect` I think we need to be careful too as there is a variable of the same name in a derived class and it's not then clear which variable is being used.